doc/dev/kdb-record.md

```diff
@@ -0,0 +1,110 @@
+
+## Exporting configuration changes
+
+`kdb record export [session_name] [--format <format>]`: Will dump the changed keys of the specified session onto `stdout`. If no session name is provided, the current session will be used. 
+
+As example, suppose the key `user:/my/app/hello` has the value `world` and you execute the following commands:
+* `kdb set user:/my/app/hello max`
+* `kdb set user:/my/app/hello susi`
+
+Only the value `susi` will be exported. Additionally, if you execute `kdb set user:/my/app/hello world` the key will not be exported at all, as there are no changes compared to the beginning of the recording.
+
+The same principle applies to the creation and deletion of keys within a session. If you create a key and delete it in the same session, it will not be exported.
+
+By default, the elektra dump format is used. The optional --format parameter can be used to specify which storage plugin should be used to format the output.
+
+## Importing configuration changes
+
+The output of the `kdb record export` command can be imported with the existing `kdb import` functionality. Depending on the specified storage plugin, other steps may be necessary, e.g. the Ansible plugins creates an Ansible playbook.
+
+## Example workflow
+
+The following example changes the IP address of www.google.com, adds an entry for www.microsoft.com and removes www.apple.com. It then shows the changes and outputs an Ansible playbook.
+
+```
+$ kdb record start AddNewHosts
+$ kdb set system:/hosts/ipv4/www.google.com 8.8.8.8
+$ kdb set system:/hosts/ipv4/www.microsoft.com 4.4.4.4
+$ kdb rm system:/hosts/ipv4/www.apple.com
+$ kdb record stop
+$ kdb record changes
+Changed system:/hosts/ipv4/www.google.com from 1.2.3.4 to 8.8.8.8
+Added system:/hosts/ipv4/www.microsoft.com with value 4.4.4.4
+Removed system:/hosts/ipv4/www.apple.com
+$ kdb record export --format ansible
+<output corresponding to ansible-libelektra plugin (https://galaxy.ansible.com/elektra_initiative/libelektra)>
+```
+
+## Architecture
+
+The core of `kdb record` consists of two parts: 
+* a global plugin that records changes to the key database
+* tooling that
+  - allows configuration of the plugin (enable, disable)
+  - allows managing recording sessions
+  - computes the minimal change set and displays and exports it 
+
+### Session storage
+
+The recording sessions and plugin configuration will be stored in Elektra. As this is a per-user feature, it will go into the `user:/` namespace. As for now, the root key is `user:/record-elektra/`.
+
+The key `user:/record-elektra/config/current_session` contains the name of the current session. If this key does not exist (or its value is empty) the recording plugin is disabled.
+
+All recording sessions are located under `user:/record-elektra/sessions/<SESSION_NAME>`.
+
+Every modification will include its UTC timestamp and the modified key in its path. 
```

Comment:
What about actions occurring within a second? Sequence numbers?

Reply:
Good question! This is only a problem if the same key was to be changed multiple times in the same time interval.
We could use a timestamp with nanosecond resolution or ticks, that should be good enough. 
Sequence numbers would also work, if we can generate them reliably.